### DISCUSSION TITLE
Suggestion: core_extensions for selected classes

### DISCUSSION THREAD

**Discussion by demuux** [jeremyevans/sequel#1873] [General]
For me, the most used method from `Sequel` module is `Sequel::lit`. I appreciate the rich DSL provided by the Sequel gem, but most of the time, if I need to write something more complex, I prefer to use short `Sequel.lit('...')` snippets.

Recently I learned about core_extensions, but I only care about the single `'...'.lit` extension.

Is it possible to load only `String` extensions? If not, can we make that happen?

**Comment by jeremyevans** [comment]
This isn't currently possible and it is not a feature I'm interested in adding.  Users who only need a subset of the core extensions should copy the code they need.  In your case, that would be:

```ruby
class String
  def lit(*args)
    args.empty? ? Sequel::LiteralString.new(self) : Sequel::SQL::PlaceholderLiteralString.new(self, args)
  end
end
```
